src/pyqasm/exceptions.py

```diff
@@ -56,17 +68,39 @@ def raise_qasm3_error(
     Args:
         message: The error message. If not provided, a default message will be used.
         err_type: The type of error to raise.
+        error_node: The QASM node that caused the error.
         span: The span (location) in the QASM file where the error occurred.
         raised_from: Optional exception from which this error was raised (chaining).
 
     Raises:
         err_type: The error type initialized with the specified message and chained exception.
     """
+    error_parts = []
+
     if span:
-        logging.error(
-            "Error at line %s, column %s in QASM file", span.start_line, span.start_column
+        error_parts.append(
+            f"Error at line {span.start_line}, column {span.start_column} in QASM file"
         )
 
+    if error_node:
+        try:
+            if isinstance(error_node, QASMNode):
+                error_parts.append("\n >>>>>> " + dumps(error_node, indent="    ") + "\n")
+            elif isinstance(error_node, list):
+                error_parts.append(
+                    "\n >>>>>> "
+                    + " , ".join(dumps(node, indent="    ") for node in error_node + "\n")
+                )
+        except Exception as _:  # pylint: disable = broad-exception-caught
+            error_parts.append("\n >>>>>> " + str(error_node))
+
+    if error_parts:
+        logger.error("\n".join(error_parts))
+
+    if os.environ.get("PYQASM_EXPAND_TRACEBACK") == "0":
```

Comment:
if named `PYQASM_EXPAND_TRACEBACK`, feels more like it should be set to a "true" or "false". Setting it equal to a number e.g. "0" or "1" would make more sense if this environment variable was named something like`PYQASM_TRACEBACKLIMIT` for consistency.